File: python/problems/242_valid_anagram.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        return self.count_chars(s) == self.count_chars(t)

    def count_chars(self, s: str):
        count: dict = dict()
        for i in range(len(s)):
            if s[i] not in count:
                count[s[i]] = 0
            count[s[i]] = count[s[i]] + 1    
        return count

File: rust/242_valid_anagram.rs
use std::collections::HashMap;
use std::hash::Hash;

fn frequencies<I, T>(xs: I) -> HashMap<T, usize>
where
  I: Iterator<Item = T>,
  T: Eq + Hash
{
  fn step<T>(mut acc: HashMap<T, usize>, x: T) -> HashMap<T, usize>
  where
    T: Eq + Hash
  {
    *acc.entry(x).or_insert(0) += 1;
    acc
  }

  xs.fold(HashMap::new(), step)
}

impl Solution {
    pub fn is_anagram(s: String, t: String) -> bool {
        frequencies(s.chars()) == frequencies(t.chars())
    }
}
